docs: improve component testing documentation (#822)
cypress-io/github-action: README.md

## Examples

> - [Basic (E2E)](#basic-e2e) testing

Reviewer: I wonder whether we should refer to it as "default" rather than "basic":

`End-to-End (default)`

Author: @MikeMcC399
Thanks for your comments! It's always good to have another pair of eyes looking at things!

The name "Basic" is historical when there was only E2E available and this example was a basic E2E example. I suggest changing the title to simply "End-to-End" in the table of contents (no "default") and at the top of the section, and leaving the actual workflow with its current name "example-basic.yml".  

I wanted to avoid taking this further and renaming "basic" at the workflow yml level as well. That would have other knock-on effects to several other places, including to the basic-pnpm workflow, which then also ought to be renamed, and to scripts, etc.

I have committed the updates and you can use the links I posted to review the rendered Markdown.

I also added a badge for Component Testing which had been previously omitted.

Author: I will resolve this thread and let you comment on the changes if necessary.

Reviewer: Thanks for the context! That seems reasonable

[![Basic example](https://github.com/cypress-io/github-action/workflows/example-basic/badge.svg?branch=master)](.github/workflows/example-basic.yml)

The workflow file [example-basic.yml](.github/workflows/example-basic.yml) shows how Cypress runs on GH Actions using Ubuntu (20 and 22), Windows, and macOS without additional OS dependencies necessary.

> The default [test type](https://on.cypress.io/guides/overview/why-cypress#Test-types) is [End-to-End (E2E) Testing](https://on.cypress.io/guides/overview/why-cypress#End-to-end). For the alternative test type see the [Component Testing](#component-testing) section below.

Reviewer: ```suggestion
The default [test type](https://on.cypress.io/guides/overview/why-cypress#Test-types) is [End-to-End (E2E) Testing](https://on.cypress.io/guides/overview/why-cypress#End-to-end). Alternatively, [Component Testing](https://on.cypress.io/guides/overview/why-cypress#Component) can be utilized by referencing the [Component Testing](#component-testing) section below.
```
